chia/full_node/coin_store.py

```diff
@@ -130,15 +130,16 @@ async def get_coins_added_at_height(self, height: uint32) -> List[CoinRecord]:
         return coins
 
     async def get_coins_removed_at_height(self, height: uint32) -> List[CoinRecord]:
-        cursor = await self.coin_record_db.execute(
-            "SELECT * from coin_record WHERE spent_index=? and spent=1", (height,)
-        )
+        cursor = await self.coin_record_db.execute("SELECT * from coin_record WHERE spent_index=?", (height,))
```

Comment:
I believe the reason this is (sometimes) slow is because sqlite picks the wrong index to look up first (i.e. `spent`), and sometimes the right index first (`spent_height`). The choice of index is arbitrary, which is explained [here](https://www.sqlite.org/queryplanner.html).

That document suggests running the sqlite command `ANALYZE`, to let the database discover which index is the better one to look up first.

I suspect, if we do this, the speedup would be even greater than this patch, since the filtering happens earlier and at a lower level.

Reply:
or maybe add another index:
```
CREATE INDEX IF NOT EXISTS coin_spent_and_index on coin_record(spent, spent_index)
```

Reply:
perhaps the simplest thing to test would be to change the original query to be:

```
"SELECT * from coin_record WHERE spent_index=? and +spent=1"
```

Which, iiuc, prevents the `spent` index from being used, (presumably) resulting in an indexed lookup on `spent_index` and then filtered by `spent=1`

(section 8.1 [here](https://www.sqlite.org/optoverview.html#covering_indices))